spec/unit/provider/package/yum_spec.rb

```diff
@@ -153,14 +153,14 @@
         allow(@yum_cache).to receive(:yum_binary=).with("yum")
         @provider = Chef::Provider::Package::Yum.new(@new_resource, @run_context)
         @provider.load_current_resource
-        expect(@provider.new_resource.package_name).to eq("testing")
+        expect(@provider.new_resource.package_name).to eq("testing.noarch")
```

Comment:
i've already got functional tests for dnf_package which are going to get merged over here:  #4894

those tests cover a lot of the bugs in the yum provider -- i tried to launch with dnf provider without all the bugginess that is here.

its a completely different approach, but the func tests could be adapted.